lib/kafka/producer.rb

```diff
@@ -300,13 +306,6 @@ def deliver_messages_with_retries(notification)
         end
       end
 
-      if @required_acks == 0
-        # No response is returned by the brokers, so we can't know which messages
-        # have been successfully written. Our only option is to assume that they all
-        # have.
-        @buffer.clear
-      end
```

Comment:
I'd actually prefer moving this block above the `if buffer_size.zero?` line. Makes the code a bit more readable.